solidity_contracts/erc20/contract.sol

```diff
@@ -1481,21 +1481,36 @@ contract ERC20Upgradeable is Initializable, ContextUpgradeable, IERC20Upgradeabl
 pragma solidity ^0.8.18;
 
 
-
 contract BridgeERC20 is Initializable, ERC20Upgradeable, UUPSUpgradeable {
-     string public linkedEOSAccountName;
+     string  eos_token_contract;
+     string  public linkedEOSAccountName;
      address public linkedEOSAddress;
      address public evmAddress;
+     uint8   public precision;
      uint256 public egressFee;
-     function initialize() initializer public {
-        __ERC20_init("Bridged USDT", "USDT");
+     function initialize(uint8   _precision,
+                         uint256 _egressFee,
+                         string memory _name, 
+                         string memory _symbol,
+                         string memory _eos_token_contract
+                          ) initializer public {
+        __ERC20_init(_name, _symbol);
         __UUPSUpgradeable_init();
         evmAddress = 0xbBBBbBbbbBBBBbbbbbbBBbBB5530EA015b900000;
         linkedEOSAddress = 0xbbBbbbBbbBBbBBbBBBbbbBbB5530eA015740a800;
         linkedEOSAccountName = "eosio.erc2o";
-        egressFee = 0.1e18;
+        precision = _precision;
+        egressFee = _egressFee;
+        eos_token_contract = _eos_token_contract;
```

Comment:
What's this for?

Reply:
I mean the eos_token_contract

Reply:
It's for the `eosTokenContract` view function which is the general pattern needed for the erc20 contract in stage 2 to validate whether it should even be making calls to the contract on behalf of some token. This doesn't really matter in stage 1 since the only implementation contracts that can be used are once deployed by the erc20 contract for each token; so we can get away with not using the `eosTokenContract` view function. But it is a useful interface to establish early.